ttnn/cpp/ttnn/operations/conv/conv2d/prepare_conv2d_weights.cpp

```diff
@@ -59,6 +42,54 @@ void validate_weights_format(const std::string& weights_format) {
     TT_ASSERT(weights_format == "OIHW", "Conv2d weights format must be \"OIHW\"");
 }
 
+template <typename T>
+bool check_non_tile_mul_width(
+    T *device,
+    const Conv2dConfig& conv_config,
+    const uint32_t in_channels
+){
+    ShardOrientation shard_orientation =
+        conv_config.transpose_shards ? ShardOrientation::COL_MAJOR : ShardOrientation::ROW_MAJOR;
+    auto num_cores_c = shard_orientation == ShardOrientation::COL_MAJOR ? device->compute_with_storage_grid_size().y : device->compute_with_storage_grid_size().x;
```

Comment:
do we really need the `shard_orientation` variable? you can directly check `transpose_shards`.